tests/OmniSharp.Roslyn.CSharp.Tests/SignatureHelpFacts.cs

```diff
@@ -474,13 +474,127 @@ public Program(Program p, int n)
     {
     }
 }";
-
             var actual = await GetSignatureHelp(source);
             Assert.Equal(3, actual.Signatures.Count());
             Assert.Equal(1, actual.ActiveParameter);
             Assert.Contains("ctor2", actual.Signatures.ElementAt(actual.ActiveSignature).Documentation);
         }
 
+        [Fact]
+        public async Task SignatureHelpForInheritedMethods()
+        {
+            const string source =
+@"public class MyBase
+{
+    public void MyMethod(int a) { }
+    public void MyMethod(int a, int b) { }
+}
+
+public class Class1 : MyBase
+{
+    public void MyMethod(int a, int b, int c) { }
+    public void MyMethod(int a, int b, int c, int d) { }
+}
+
+public class Class2
+{
+    public void foo()
+    {
+        Class1 c1 = new Class1();
+        c1.MyMethod($$);
+    }
+
+ }";
+            var actual = await GetSignatureHelp(source);
+            Assert.Equal(4, actual.Signatures.Count());
+        }
+
+        [Fact]
+        public async Task SignatureHelpForInheritedInaccesibleMethods()
```

Comment:
I'd like to see a few more tests for variants of this scenario:

1. An inherited protected method appears in signature help within the a sub type.

    ```C#
    class A
    {
        protected void M1() { }
    }

    class B : A
    {
        void M2()
        {
            $$ // Both M1() and M2()
        }
    }
    ```

2. An inherited protected method appears in signature help after `this.`.

    ```C#
    class A
    {
        protected void M1() { }
    }

    class B : A
    {
        void M2()
        {
            this.$$ // Both M1() and M2()
        }
    }
    ```

3. An inherited protected method appears in signature help after `base.`.

    ```C#
    class A
    {
        protected void M1() { }
    }

    class B : A
    {
        void M2()
        {
            base.$$ // Only M1()
        }
    }
    ```

4. Only static methods appear in a static context.

    ```C#
    class A
    {
        protected static void M1() { }
        public void M2() { }
    }

    class B : A
    {
        static void M3()
        {
            A.$$ // Only M1()
        }
        public void M4() { }
    }
    ```

    ```C#
    class A
    {
        protected static void M1() { }
        public void M2() { }
    }

    class B : A
    {
        static void M3()
        {
            B.$$ // Both M1() and M3()
        }
        public void M4() { }
    }
    ```

5. Static and instance methods appear in an instance context.

    ```C#
    class A
    {
        protected static void M1() { }
        public void M2() { }
    }

    class B : A
    {
        static void M3()
        {
            $$ // M1(), M2(), M3() and M4() appear
        }
        public void M4() { }
    }
    ```